tempodb/tempodb.go

```diff
@@ -531,3 +497,50 @@ func includeCompactedBlock(c *backend.CompactedBlockMeta, id common.ID, blockSta
 	}
 	return includeBlock(&c.BlockMeta, id, blockStart, blockEnd, timeStart, timeEnd)
 }
+
+// createLegacyCache uses the config to return a cache and a list of roles.
+func createLegacyCache(cfg *Config, logger gkLog.Logger) (cache.Cache, []cache.Role, error) {
```

Comment:
Instead of this, we could translate the legacy config to the new cache config in `modules.go` when the module is instantiated. And we wouldn't need `AddCache`.

Reply:
Point taken, but I'm going to leave this where it is. It's legacy code that we will be removing in a version or 2 and it will feel like a wart no matter where we add it.